keras/engine/training.py

```diff
@@ -3753,6 +3780,28 @@ def _should_eval(self, epoch, validation_freq):
                 f"type {type(validation_freq)}."
             )
 
+    @property
+    def states(self):
+        """Returns the list of all layer states.
```

Comment:
I think we need to make it clear that the states are from stateful layers and are non-trainable variable.